src/js/editor/post/post-inserter.js

```diff
@@ -0,0 +1,303 @@
+
+    preItem  = this.builder.createListItem();
+    postItem = this.builder.createListItem();
+
+    let prevMarker = item.markerBeforeOffset(this.cursorOffset);
+    appendPre = true;
+    item.markers.forEach(marker => {
+      if (appendPre) {
+        preItem.markers.append(marker.clone());
+        if (marker === prevMarker) {
+          appendPre = false;
+        }
+      } else {
+        postItem.markers.append(marker.clone());
+      }
+    });
+
+    this._replaceSection(item, [preItem, postItem]);
+
+    let blank    = this.builder.createMarkupSection();
+    let preList  = this.builder.createListSection(list.tagName);
+    let postList = this.builder.createListSection(list.tagName);
+
+    appendPre = true;
+    list.items.forEach(_item => {
+      if (appendPre) {
+        preList.items.append(_item.clone());
+        if (_item === preItem) {
+          appendPre = false;
+        }
+      } else {
+        postList.items.append(_item.clone());
+      }
+    });
+
+    this._replaceSection(list, [preList, blank, postList]);
+    return [preList, blank, postList];
+  }
+
+  _insertSectionAfter(section, parent) {
+    assert('Cannot _insertSectionAfter nested section', !parent.isNested);
+    let reference = parent.next;
+    let collection = this._post.sections;
+    this.postEditor.insertSectionBefore(collection, section, reference);
+  }
+
+  _wrapNestedSection(section) {
+    let tagName = section.parent.tagName;
+    let parent = this.builder.createListSection(tagName);
+    parent.items.append(section.clone());
+    return parent;
+  }
+
+  _mergeSection(section) {
+    assert('Can only merge markerable sections',
+           this._isMarkerable && section.isMarkerable);
+    this._hasInsertedFirstLeafSection = true;
+
+    let markers = section.markers.map(m => m.clone());
+    let position = this.postEditor.insertMarkers(this.cursorPosition, markers);
+
+    this.cursorPosition = position;
+  }
+
+  // Can be called to add a line break when in a nested section or a parent
+  // section.
+  _breakAtCursor() {
+    if (this.cursorSection.isBlank) {
+      return;
+    } else if (this._isMarkerable) {
+      this._breakMarkerableAtCursor();
+    } else {
+      this._breakNonMarkerableAtCursor();
+    }
+  }
+
+  // inserts a blank section before/after the cursor, depending on cursor position
+  _breakNonMarkerableAtCursor() {
+    let collection = this._post.sections,
+        blank = this.builder.createMarkupSection(),
+        reference = this.cursorPosition.isHead() ? this.cursorSection :
+                                                   this.cursorSection.next;
+    this.postEditor.insertSectionBefore(collection, blank, reference);
+    this.cursorPosition = blank.tailPosition();
+  }
+
+  _breakMarkerableAtCursor() {
+    let [pre, post] = this.cursorSection.splitAtPosition(this.cursorPosition)
+                          .map(s => s.clone());
+    this.postEditor._coalesceMarkers(pre);
```

Comment:
I think `_coalesceMarkers` is an implementation detail that no consumer should have a need to use. It was in use here because this code breaks a section without going through the postEditor. I changed it to use `postEditor#splitSections` instead